### ISSUE TITLE
Built-in Clock Widget doesnt respect the LC_TIME variable

### ISSUE THREAD

**Issue by parham-dev27** [qtile/qtile#4353]
### The issue:

The LC_TIME variable which defines the time locale, is not used by the clock widget. So if for example $LC_TIME="de_DE.UTF-8", but the widget still shows the month/day names in english. 

I am able to fix this, if you want let me know. I can add a pr for it. Thank you

### Required:

- [x] I have searched past issues to see if this bug has already been reported.

**Comment by elParaguayo** [comment]
Shouldn't the locale be systemwide? You can just add these lines to your config:
```python
import locale
locale.setlocale(locale.LC_TIME, "de_DE.UTF-8")
```
And that should work.

I'm surprised python's `datetime` module isn't picking it up be default.

**Comment by parham-dev27** [comment]
Thank you. I also created a pull request fixing this.

**Comment by elParaguayo** [comment]
Did some digging. This is from the python docs:
```
Initially, when a program is started, the locale is the C locale, no matter what the user’s preferred locale is. There is one exception: the [LC_CTYPE](https://docs.python.org/3/library/locale.html#locale.LC_CTYPE) category is changed at startup to set the current locale encoding to the user’s preferred locale encoding. The program must explicitly say that it wants the user’s preferred locale settings for other categories by calling setlocale(LC_ALL, '').
```
So, this is why `LC_TIME` is ignored and we do need to set it.

I'm still on the fence as to whether this needs to be in the clock widget or whether users should just set it in their configs.

**Comment by parham-dev27** [comment]
Well, you are right that users could add it to their configs, but the average user (at least for me) expects it to work out of the box. So if it doesn't work, they won't do too much digging. Alternatively, we could add this to the configuration options.

**Comment by elParaguayo** [comment]
I agree with your point about the average user and so it feels like it should be enabled by default. Let me look at the PR later.

**Comment by parham-dev27** [comment]
Fixed by #4356
